src/v/cluster/tm_stm.cc

```diff
@@ -62,10 +63,102 @@ tm_stm::tm_stm(
   , _transactional_id_expiration(
       config::shard_local_cfg().transactional_id_expiration_ms.bind())
   , _feature_table(feature_table)
-  , _ctx_log(logger, ssx::sformat("[{}]", _raft->ntp())) {}
+  , _ctx_log(logger, ssx::sformat("[{}]", _raft->ntp())) {
+    setup_metrics();
+}
 
 ss::future<> tm_stm::start() { co_await persisted_stm::start(); }
 
+void tm_stm::setup_metrics() {
+    namespace sm = ss::metrics;
+
+    auto ns_label = sm::label("namespace");
+    auto topic_label = sm::label("topic");
+    auto partition_label = sm::label("partition");
```

Comment:
maybe it would be enough toi label coordinator with partition id ?, the topic and namespace are always the same for the tx coordinator